modules/@apostrophecms/schema/ui/apos/logic/AposInputArray.js

```diff
@@ -124,16 +175,66 @@ export default {
     }
   },
   methods: {
+    toggleAll(open) {
+      this.items = this.items.map(item => ({
+        ...item,
+        open
+      }));
+    },
+    startDragging(event) {
+      this.isDragging = true;
+      this.disengageAll();
+      this.toggleEngage({ target: event.item });
+    },
+    stopDragging(event) {
+      this.isDragging = false;
+      document.getSelection().removeAllRanges();
+      this.focusElement(event.item.getAttribute('data-id'));
+    },
+    getInlineMenuItems(index) {
+      const menu = klona(this.inlineContextMenu);
+      if (index === 0 && menu.find(i => i.action === 'move-up')) {
+        menu.find(i => i.action === 'move-up').modifiers = [ 'disabled' ];
+      }
+      if (index + 1 === this.items.length && menu.find(i => i.action === 'move-down')) {
+        menu.find(i => i.action === 'move-down').modifiers = [ 'disabled' ];
+      }
+      return menu;
+    },
+    getTableHeaderClass(field, baseClass) {
+      let label = this.$t(field.label);
+      const validChars = /[^a-zA-Z0-9_-]/g;
+      label = label.replace(validChars, '-');
+      label = label.toLowerCase();
+      return `${baseClass}--${label}`;
+    },
+    toggleEngage(event, options) {
+      let elId = event.target.getAttribute('data-id');
+      if (!elId && !options.exact) {
+        elId = event.target.closest('[data-id]').getAttribute('data-id');
+      }
+      if (elId) {
+        if (this.items.find(i => i._id === elId)) {
+          const was = this.items.find(i => i._id === elId).engaged;
+          this.disengageAll();
+          this.items.find(i => i._id === elId).engaged = !was;
+        } else {
+          this.disengageAll();
+        }
+      }
+    },
```

Comment:
Optional: Early return to avoid if nesting:
```javascript
    toggleEngage(event, options) {
      const dataIdAttr = event.target.getAttribute('data-id');
      const elId = !dataIdAttr && !options.exact
        ? event.target.closest('[data-id]').getAttribute('data-id')
        : dataIdAttr;
      if (elId) {
        return;
      }

      if (this.items.find(i => i._id === elId)) {
        const was = this.items.find(i => i._id === elId).engaged;
        this.disengageAll();
        this.items.find(i => i._id === elId).engaged = !was;
      } else {
        this.disengageAll();
      }
    },

```